From rjw@sisk.pl  Mon Mar 26 14:57:30 2012
From: "Rafael J. Wysocki" <rjw@sisk.pl>
Date: Mon, 26 Mar 2012 23:24:40 +0200
Subject: PM / Domains: Check domain status during hibernation restore of devices
To: Greg KH <gregkh@linuxfoundation.org>
Cc: horms@verge.net.au, lethal@linux-sh.org, stable@vger.kernel.org, stable-commits@vger.kernel.org
Message-ID: <201203262324.40855.rjw@sisk.pl>


From: Rafael J. Wysocki <rjw@sisk.pl>

This is a backport of mainline commit
18dd2ece3cde14cfd42e95a89eb14016699a5f15.

Power domains that were off before hibernation shouldn't be turned on
during device restore, so prevent that from happening.

This change fixed up mainline commit
65533bbf63b4f37723fdfedc73d0653958973323

    PM / Domains: Fix hibernation restore of devices, v2

that didn't include it by mistake.

Signed-off-by: Rafael J. Wysocki <rjw@sisk.pl>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

---
 drivers/base/power/domain.c |    3 +++
 1 file changed, 3 insertions(+)

--- a/drivers/base/power/domain.c
+++ b/drivers/base/power/domain.c
@@ -1064,6 +1064,9 @@ static int pm_genpd_restore_noirq(struct
 		}
 	}
 
+	if (genpd->suspend_power_off)
+		return 0;
+
 	pm_genpd_poweron(genpd);
 	genpd_start_dev(genpd, dev);
 
